src/service/scheduler.js

```diff
@@ -180,13 +239,14 @@ export class Scheduler {
    */
   observed_(entries) {
     for (let i = 0; i < entries.length; i++) {
-      const {target, isIntersecting} = entries[i];
+      const {target, isIntersecting: isThisIntersecting} = entries[i];
 
       const current = this.targets_.get(target);
       if (!current) {
         continue;
       }
 
+      const isIntersecting = isThisIntersecting || current.isIntersecting;
       this.targets_.set(target, {asap: current.asap, isIntersecting});
       if (isIntersecting) {
         this.maybeBuild_(target);
```

Comment:
Nit: as a performance optimization, we can simply skip the `set` if `!isThisIntersecting` instead of trying to merge.

```suggestion
      if (isThisIntersecting) {
        this.targets_.set(target, {asap: current.asap, isIntersecting: isThisIntersecting});
        this.maybeBuild_(target);
```